src/Network/Tox/Application/GroupChatsPackets.lhs

```diff
@@ -0,0 +1,654 @@
+\chapter{DHT Group Chats Packet Protocols}
+
+All packet fields are considred mandatory unless flagged as \textbf{\verb'[optional]'}.
+The minimum size of a lossless packet is 86 bytes, and a lossy packet 78 bytes.
+The maximum size of a packet is 1400 bytes.
+
+\section{Full Packet Structure}
+
+\subsection{Plaintext header}
+
+\begin{tabular}{l|l}
+  Length             & Contents \\
+  \hline
+  \texttt{1}         & Packet Kind \\
+  \texttt{4}         & Chat ID Hash \\
+  \texttt{32}        & Public Encryption Key \\
+  \texttt{24}        & Nonce \\
+\end{tabular}
+
+\subsection{Encrypted header}
+
+\begin{tabular}{l|l}
+  Length             & Contents \\
+  \hline
+  \texttt{0-8}       & Padding \\
+  \texttt{1}         & Group Packet Identifier \\
+  \texttt{8}         & Message Id \textbf{\verb'[optional]'} \\
+\end{tabular}
+
+\subsection{Encrypted payload}
+\begin{tabular}{l|l}
+  Length             & Contents \\
+  \hline
+  Variable           & Payload \\
+\end{tabular}
+
+The plaintext header contains a \textbf{\verb'Toxcore Network Packet Kind'}
+which identifies the toxcore networking level packet type. These types
+are:
+\begin{tabular}{l|l}
+  Type                                    & Net Packet ID \\
+  \hline
+  \textbf{\verb'NET_PACKET_GC_HANDSHAKE'} & 0x5a \\
+  \textbf{\verb'NET_PACKET_GC_LOSSLESS'}  & 0x5b \\
+  \textbf{\verb'NET_PACKET_GC_LOSSY'}     & 0x5c \\
+\end{tabular}
+
+The \textbf{\verb'Chat ID Hash'} is a
+\href{https://en.wikipedia.org/wiki/Jenkins_hash_function#one_at_a_time}{Jenkins One At A Time Hash}
+of the chat ID. This is used to identify which group a particular message is intended
+for. The \textbf{\verb'sender public encryption key'} is used to identify the peer
+who sent the packet, and the \textbf{\verb'nonce'} is used for decryption.
+
+The encrypted header for lossless and lossy packets contains between 0
+and 8 bytes of empty padding, which is used to mitigate certain types of
```

Comment:
Explain how the number between 0 and 8 is computed. Also specify which crypto attacks it mitigates.

Reply:
This was copied from https://github.com/TokTok/c-toxcore/blob/master/toxcore/net_crypto.c#L1100

I can't remember the purpose of the padding, and it's not documented elsewhere in the spec.